ext/pdo_mysql/mysql_sql_parser.re

```diff
@@ -0,0 +1,48 @@
+/*
+  +----------------------------------------------------------------------+
+  | Copyright (c) The PHP Group                                          |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | https://www.php.net/license/3_01.txt                                 |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Matteo Beccati <mbeccati@php.net>                            |
+  +----------------------------------------------------------------------+
+*/
+
+
+#include "php.h"
+#include "pdo/php_pdo_driver.h"
+#include "pdo/php_pdo_int.h"
+#include "pdo/pdo_sql_parser.h"
+
+int pdo_mysql_scanner(pdo_scanner_t *s)
+{
+	const char *cursor = s->cur;
+
+	s->tok = cursor;
+	/*!re2c
+	BINDCHR		= [:][a-zA-Z0-9_]+;
+	QUESTION	= [?];
+	COMMENTS	= ("/*"([^*]+|[*]+[^/*])*[*]*"*/"|("--"|[#])[^\r\n]*);
```

Comment:
see https://github.com/atk4/data/blob/13815355bb/src/Persistence/Sql/Mysql/Expression.php#L19

1. `--` does not have to be a comment if followed by non-ws character
2. the `[^...]` must not contain `\r`

Reply:
I tried in so many ways to achieve that goal, but alas adding a space somehow breaks the parser. I reached the end of re2c skills and I gave up. The limitation was in place even before this RFC and I plan to add a comment about it in the next round of editing the RFC.

Reply:
Fixed in ad55a61